zerver/tests/test_messages.py

```diff
@@ -3528,6 +3522,24 @@ def test_claim_attachment(self) -> None:
             attachment = Attachment.objects.get(path_id=path_id)
             self.assertTrue(attachment.is_claimed())
 
+class LinkDetectionTest(ZulipTestCase):
+    def test_finds_all_links(self) -> None:
+        msg_ids = []
+        msg_contents = ["foo.org", "[bar](baz.gov)", "http://quux.ca"]
+        for msg_content in msg_contents:
+            msg_ids.append(self.send_stream_message(self.example_email('hamlet'),
+                                                    'Denmark', content=msg_content))
+        msgs = [Message.objects.get(id=id) for id in msg_ids]
+        self.assertTrue(all([msg.has_link for msg in msgs]))
```

Comment:
We should also add some negative tests here, where we shouldn't detect a link.

Example:

1. plain sentence without link.
2. Link in codeblock `https://google.com`.
3. (..)

Reply:
Agree, added link in inline code, link in code block, link in LaTeX, and plain text.